src/js/jsx/search/SearchBar.jsx

```diff
@@ -142,6 +193,36 @@ define(function (require, exports, module) {
         },
 
         /**
+         * Get the layer type
+         *
+         * @private
+         * @param {Layer} layer
+         * @return {Array.<string>}
+         */
+        _getLayerType: function (layer) {
+            var layerType = ["layer"];
+            _.forEach(Object.keys(layer.layerKinds), function (kind) {
```

Comment:
That is to say, I'm either confused about what this function is for, or it should actually return just a single kind? If not, then this information ought to be static and computed only once?